### PR TITLE
Select: Add filterFunction prop

### LINKED ISSUE #444: Select.Options unable to display accented characters
**Description:**

When the search term is unaccented and the fetch results have labels with accented characters, these options are filtered out as the initial search term is unaccented.

**Suggested Solution:**

Allow Select to accept a custom filter function, eg.

`filterOptions(options, searchTerm) {
    ...
}`

**Ref:** 
https://gitlab.glints.com/glints/glints-dst/-/merge_requests/1779

**Side investigation:** 
Select.Option takes children as any[] instead of string 
https://gitlab.glints.com/glints/glints-dst/-/merge_requests/1779#note_52508

### PR DESCRIPTION
**Pull request by westwood846**
Closes #444. Solution is to provide a prop for specifying a custom filter function. For the accented characters problem, such a function can transform the option string and the search string to remove the accent characters as required.

### CODE REVIEW

File: stories/Input/SelectStory.tsx [hunk 1]
```diff
@@ -139,6 +139,17 @@ const props = {
       require: 'no',
       description: 'Sets initial open state of Select.',
     },
+    {
```
Comment (ritikamotwani) [inline stories/Input/SelectStory.tsx:142]: @westwood846 , could you add a usage for this prop in the examples. 
Also the prop type isn't boolean, right?

File: src/Input/Select/Select.test.tsx [hunk 2]
```diff
@@ -53,6 +55,27 @@ function setupOpenSelectMenu() {
...
 
+function setupWithCustomFilterFunction(
+  filterFunction: (option: string, search: string) => boolean
+) {
+  const { getByRole, queryAllByTestId, getByTestId } = render(
+    SelectComponentWithExtraProps({ filterFunction })
+  );
+  const selectInput = getByRole('combobox') as HTMLInputElement;
+  const selectList = getByTestId('listbox');
+  // const selectLabel = getByTestId('select-label');
```
Comment (ritikamotwani) [inline src/Input/Select/Select.test.tsx]: Why is this comment here?

File: src/Input/Select/Select.tsx [hunk 3]
```diff
@@ -116,11 +117,13 @@ const Select: ISelect = (props: Props) => {
     }
 
     // for search results, so we do toLower to filter matched options
```
Comment (ritikamotwani) [inline src/Input/Select/Select.tsx]: This comment should be changed now?

File: stories/Input/SelectStory.tsx (hunk 1 above)
Comment (westwood846) [reply stories/Input/SelectStory.tsx:142]: Added an example and updated the prop here.

File: src/Input/Select/Select.test.tsx (hunk 2 above)
Comment (westwood846) [reply src/Input/Select/Select.test.tsx]: Removed

File: src/Input/Select/Select.tsx (hunk 3 above)
Comment (westwood846) [reply src/Input/Select/Select.tsx]: Removed the whole comment, not really helpful anyway

**Review by ritikamotwani — APPROVED** [review]
LGTM

**Pull request merged**